agent/src/com/cloud/agent/Agent.java

```diff
@@ -699,16 +719,97 @@ public void processResponse(final Response response, final Link link) {
         }
     }
 
-    public void processReadyCommand(final Command cmd) {
+    /**
+     * Checks if received list is different to actual management server list (in order and size)
+     * @param actual actual list
+     * @param received received list
+     * @return true if list is updated, false if not
+     */
+    private boolean isReceivedListUpdated(List<String> actual, List<String> received) {
+        if (actual.size() != received.size()) {
+            return true;
+        }
+        for (int i = 0; i < received.size(); i++) {
```

Comment:
We can simply using the previous implementation to simply check if the existing setting differs in size or `set`.